src/typecheck/mod.rs

```diff
@@ -1104,28 +1262,7 @@ fn row_add(
 }
 
 /// Try to unify two types.
-///
-/// A wrapper around `unify_` which just checks if `strict` is set to true. If not, it directly
-/// returns `Ok(())` without unifying anything.
-pub fn unify(
-    state: &mut State,
-    strict: bool,
-    t1: TypeWrapper,
-    t2: TypeWrapper,
-) -> Result<(), UnifError> {
-    if strict {
-        unify_(state, t1, t2)
-    } else {
-        Ok(())
-    }
-}
-
-/// Try to unify two types.
-pub fn unify_(
-    state: &mut State,
-    mut t1: TypeWrapper,
-    mut t2: TypeWrapper,
-) -> Result<(), UnifError> {
+pub fn unify(state: &mut State, mut t1: TypeWrapper, mut t2: TypeWrapper) -> Result<(), UnifError> {
     if let TypeWrapper::Ptr(pt1) = t1 {
         t1 = state.table.root(pt1);
     }
```

Comment:
Does this function really update t1?

Reply:
Yes? Why not? It's a mutable variable in scope. If it wouldn't, then what `t1` would it update? (Also, if it wouldn't, most of the typechecking process would probably be utterly broken). On the other hand, I personally don't really like this way of having a mutable parameter (I would shadow `t1` with another `t1` both being immutable). But this code wasn't introduce by this PR, so if we prettify it, let's do that in another one :slightly_smiling_face: